src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1213,10 +1213,17 @@ def _need_update_container(self):
 
     def set_up_existing_container_update(self):
         if self.get_argument_force_single_container_updates():
+            containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])
+            if not containers or len(containers) == 0:
```

Comment:
You have use `default=[]` at
`containers = safe_get(self.new_containerapp, "properties", "template", "containers", default=[])`

No need `not containers`